zerver/lib/markdown/__init__.py

```diff
@@ -1616,23 +1627,35 @@ def prepare_realm_pattern(source: str) -> str:
     OUTER_CAPTURE_GROUP."""
     return fr"""(?<![^\s'"\(,:<])(?P<{OUTER_CAPTURE_GROUP}>{source})(?!\w)"""
 
+# Convert user inputted %()s realm filter strings to be {} type strings
+def convert_to_format_string(percent_string: str) -> str:
+    regex = RealmFilter.group_match_regex
+    return regex.sub(lambda m: "{{{}}}".format(m.group('group_name')), percent_string)
```

Comment:
This will break on `%`-formatting strings containing `%%`, `{`, or `}`, and calls for more principled parsing than regexes can reasonably provide.